examples/graphene_apps/python_worker/avalon_worker/tests/test_send_work_order.py

```diff
@@ -0,0 +1,178 @@
+        replymessage = "Error while processing work order"
+        logger.error("Exception: {} args {} details {}"
+                     .format(type(ex), ex.args, ex))
+    return replymessage
+
+
+def _create_json_request(method_name, params):
+    json_request = {}
+    json_request["jsonrpc"] = "2.0"
+    json_request["id"] = random.randint(0, 100000)
+    json_request["method"] = method_name
+    json_request["params"] = params
+    return json_request
+
+
+def _generate_session_key_iv():
+    encrypt = worker_encryption.WorkerEncrypt()
+    session_key = encrypt.generate_session_key()
+    session_key_iv = encrypt.generate_iv()
+    return (session_key, session_key_iv)
+
+
+def _create_work_order_request(workload_id, input_data_str,
+                               session_key, session_key_iv,
+                               worker_signup_json):
+    worker_enc_key_str = worker_signup_json["encryption_key"]
+
+    encrypt = worker_encryption.WorkerEncrypt()
+    encrypted_session_key = encrypt.encrypt_session_key(
+        session_key, worker_enc_key_str)
+
+    msg = input_data_str
+    msg_bytes = msg.encode("UTF-8")
+    encrypt_msg_bytes = encrypt.encrypt_data(msg_bytes,
+                                             session_key, session_key_iv)
+    input_json = dict()
+    input_json["jsonrpc"] = "2.0"
+    input_json["method"] = "WorkOrderSubmit"
+    input_json["id"] = "1"
+    input_json["params"] = dict()
+    input_json["params"]["encryptedSessionKey"] = encrypted_session_key.hex()
+    if session_key_iv:
+        input_json["params"]["sessionKeyIv"] = session_key_iv.hex()
+    input_json["params"]["workloadId"] = workload_id.encode("UTF-8").hex()
+    input_json["params"]["workOrderId"] = secrets.token_hex(32)
+    worker_id = hashlib.sha256("graphene-hello".encode("UTF-8")).hexdigest()
+    input_json["params"]["workerId"] = worker_id
+    input_json["params"]["requesterId"] = secrets.token_hex(32)
+    input_json["params"]["requesterNonce"] = secrets.token_hex(32)
+
+    in_data = dict()
+    in_data["index"] = 0
+    in_data["dataHash"] = ""
+    in_data["data"] = msg_bytes
+    in_data["encryptedDataEncryptionKey"] = "null"
+    in_data["iv"] = ""
+    input_json["params"]["inData"] = [in_data]
+
+    # Encrypt inData
+    encrypt.encrypt_work_order_data_json(input_json["params"]["inData"],
+                                         session_key, session_key_iv)
+
+    req_hash = worker_hash.WorkerHash().calculate_request_hash(
+        input_json["params"])
+    encrypted_req_hash = encrypt.encrypt_data(req_hash,
+                                              session_key, session_key_iv)
+    input_json["params"]["encryptedRequestHash"] = encrypted_req_hash.hex()
+
+    return json.dumps(input_json)
+
+
+def _get_work_order_result(response_json, session_key, session_key_iv):
+    encrypt = worker_encryption.WorkerEncrypt()
+    encrypt.decrypt_work_order_data_json(response_json["result"]["outData"],
```

Comment:
What if work order failed and it has error tag?

Reply:
We are checking if "result" is present in response before calling this function